### DISCUSSION TITLE
outline can't use psql or redis on cluster deploys

### DISCUSSION THREAD

**Discussion by thatguyatgithub** [outline/outline#5592] [Self Hosting]
I attemped to create an Outline setup that uses both pgsql and redis running on clusters setups to maintain HA on a highly volatile kubernetes environment.
The problem seems that outline doesn't know when arriving to a replica that is an slave, and attempts to run RW operations, which is both true for redis and pgsql.

I'm afraid the way in which connections are declared doesn't make transparent to Outline that there's a service LB to which the backend (redis or pgsql) are possible to be replicas, which randomly raises errors when atempting to run RW commands on a replica, as expected
```
ReplyError: READONLY You can't write against a read only replica.
    at parseError (/opt/outline/node_modules/redis-parser/lib/parser.js:179:12)
    at parseType (/opt/outline/node_modules/redis-parser/lib/parser.js:302:14) {
  command: { name: 'del', args: [ 'UPDATES_KEY' ] }
```

Is there any way to circunvent this from Outline itself?

**Comment by tommoor** [comment]
> Is there any way to circunvent this from Outline itself?

No, just keep the setup simple.
